pkg/cluster/resources.go

```diff
@@ -221,6 +219,23 @@ func (c *Cluster) deleteConnectionPooler() (err error) {
 
 	c.logger.Infof("Connection pooler service %q has been deleted", serviceName)
 
+	// Repeat the same for the secret object
+	secretName := c.credentialSecretName(c.OpConfig.ConnectionPooler.User)
+
+	secret, err := c.KubeClient.
+		Secrets(c.Namespace).
+		Get(context.TODO(), secretName, metav1.GetOptions{})
+
+	if err != nil {
+		c.logger.Debugf("could not get connection pooler secret %q: %v", secretName, err)
+	} else {
+		uid := secret.UID
+		if err = c.deleteSecret(uid, *secret); err != nil {
+			return fmt.Errorf("could not delete pooler secret: %v", err)
```

Comment:
does this prevent deletion of e.g deployment? disable pooler should cleanup as good as it gets.

Reply:
nope. 1. Deployment, 2. Service, 3. Secret

It's true, however, that if 1. or 2. return an error the rest is not deleted.